internal/k8s/controller.go

```diff
@@ -792,6 +792,11 @@ func (lbc *LoadBalancerController) syncPolicy(task task) {
 		return
 	}
 
+	if !lbc.HasCorrectIngressClass(obj) {
```

Comment:
another corner case for the existing code:
```
kubectl apply -f policy-other-ingress-class.yaml
kubectl apply -f virtual-server-policy.yaml
kubectl describe vs
  Message:  Configuration for default/virtual-server was added or updated
  Reason:   AddedOrUpdated
  State:    Valid
. . .
Events:
  Type    Reason          Age   From                      Message
  ----    ------          ----  ----                      -------
  Normal  AddedOrUpdated  9s    nginx-ingress-controller  Configuration for default/virtual-server was added or updated
```
Expected:
```
  Message:  Configuration for default/virtual-server was added or updated ; with warning(s): Policy default/rate-limit-primary is missing or invalid
  Reason:   AddedOrUpdatedWithWarning
  State:    Warning
Events:
  Type     Reason                     Age   From                      Message
  ----     ------                     ----  ----                      -------
  Warning  AddedOrUpdatedWithWarning  5s    nginx-ingress-controller  Configuration for default/virtual-server was added or updated ; with warning(s): Policy default/rate-limit-primary is missing or invalid
  ```

Reply:
updated